slip-0019.md

```diff
@@ -5,24 +5,182 @@ Number:  SLIP-0019
+
+A CoinJoin coordinator can also benefit from such a proof to verify that the CoinJoin participant is able and willing to sign the input. This verification helps to mitigate denial-of-service attacks as the attacker has to use a limited UTXO set that they control and in case of misbehavior this UTXO set gets banned.
+
+## Proof of ownership format
+
+A proof of ownership consists of a proof body and a signature. The proof body contains one or more ownership identifiers which allow a wallet to efficiently determine whether or not it is able to spend a UTXO having a given scriptPubKey. The proof signature affirms that the proof body can be trusted to have been generated by the true owner of the UTXO.
+
+```
+proofOfOwnership = proofBody || proofSignature
+```
+
+### Ownership identifier
+
+Let *k* be a secret *ownership identification key* derived from the wallet's master secret using the [SLIP-0021](https://github.com/satoshilabs/slips/blob/master/slip-0021.md) method for hierarchical derivation of symmetric keys as:
+
+```
+k = Key(m/"SLIP-0019"/"Ownership identification key")
+```
+
+The ownership identifier for a scriptPubKey is computed as:
+
+```
+id = HMAC-SHA256(key = k, msg = scriptPubKey)
+```
+
+In case of *m*-of-*n* multi-signature scriptPubKeys the proof of ownership SHOULD contain the ownership identifiers of all *n* co-owners of that scriptPubKey. See [Identifier inclusion](#identifier-inclusion) for further details.
+
+A wallet MUST NOT produce and reveal an ownership identifier for a scriptPubKey which it does not control. Such a fake ownership identifier can be used to mount a denial-of-service attack.
+
+### Proof body
+
+The *proofBody* is a concatenation of the following fields:
+
+* *version* (4 bytes): b"\x53\x4c\x00\x19".
+* *flags* (1 byte, bit 0 is the least significant bit):
+  * Bit 0: User confirmation
+    * 0 means the proof was generated without user confirmation.
+    * 1 means the user confirmed the generation of the proof.
+  * Bits 1 to 7: Reserved for future use (all must be 0).
+* *n* (VarInt): the number of ownership identifiers which follow.
+* *id*<sub>1</sub> || *id*<sub>2</sub> || ... || *id*<sub>*n*</sub> (32 bytes each): concatenation of the ownership identifiers for the given scriptPubKey, one for each co-owner, see [Identifier inclusion](#identifier-inclusion) for further details.
+
+### Proof footer
+
+The *proofFooter* is a concatenation of the following fields:
+
+* *scriptPubKey* (length-prefixed string).
+* *commitmentData* (length-prefixed string), any additional data to which the proof should commit, see below.
+
+The proof footer is not a part of the proof of ownership. It is included only in the *sighash* computation. Variable-length fields are encoded the same way as in Bitcoin transactions, as a length-prefixed string, where the length is encoded as a variable-length integer (VarInt).
```

Comment:
> It is included only in the *sighash* computation.

Why? A brief justification should be in the text.

Reply:
Done in 2f84a7a.